editor/editor_file_dialog.cpp

```diff
@@ -1104,14 +1104,24 @@ void EditorFileDialog::set_access(Access p_access) {
 }
 
 void EditorFileDialog::invalidate() {
-	if (is_visible()) {
-		update_file_list();
-		_update_favorites();
-		_update_recent();
-		invalidated = false;
-	} else {
-		invalidated = true;
+	if (!is_visible() || is_invalidating) {
+		return;
+	}
+
+	is_invalidating = true;
+	call_deferred("_invalidate");
```

Comment:
You should be able to do this without using strings and binding the method.

```suggestion
	callable_mp(this, &EditorFileDialog::_invalidate).call_deferred();
```

Reply:
Same for `FileDialog`, of course.

Reply:
Did it for those, and some others that were already there.